<launch>
  <arg name="cost_map" default="true"/>
  <arg name="tag_detector" default="true"/>
  <arg name="object_detector" default="true"/>

  <node if="$(arg tag_detector)"
        pkg="nodelet" type="nodelet" name="zed_tag_detector"
        args="load mrover/TagDetectorNodelet perception_nodelet_manager" output="screen"/>

  <node if="$(arg tag_detector)"
        pkg="nodelet" type="nodelet" name="long_range_tag_detector"
        args="load mrover/LongRangeTagDetectorNodelet perception_nodelet_manager" output="screen"/>

  <node if="$(arg object_detector)"
        pkg="nodelet" type="nodelet" name="object_detector" respawn="true"
        args="load mrover/ObjectDetectorNodelet perception_nodelet_manager" output="screen"/>

  <node if="$(arg cost_map)"
        pkg="nodelet" type="nodelet" name="cost_map"
        args="load mrover/CostMapNodelet perception_nodelet_manager" output="screen"/>
</launch>
